ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -604,16 +628,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
     renderPreviewDataTable() {
       const filteredPreviewData =
           this.state.previewList.filter(
-            preview => fp.contains(preview.domain, this.state.selectedPreviewDomain))[0];
-      return <DataTable ref={el => this.dt = el} key={this.state.selectedPreviewDomain}
-                        scrollable={true} style={{width: '100%'}}
-                        value={this.getDataTableValue(filteredPreviewData.values)}>
-        {filteredPreviewData.values.map(value =>
-          <Column key={value.value} header={this.getHeaderValue(value)}
-                  headerStyle={{textAlign: 'left', width: '5rem'}} style={{width: '5rem'}}
-                  field={value.value}/>
-        )}
-      </DataTable>;
+            preview => preview.domain === this.state.selectedPreviewDomain)[0];
+      const domainDisplayed = this.state.selectedPreviewDomain.toString().toLowerCase();
+      return filteredPreviewData.values.length > 0 ?
+        <DataTable ref={el => this.dt = el} key={this.state.selectedPreviewDomain}
+                          scrollable={true} style={{width: '100%'}}
+                          value={this.getDataTableValue(filteredPreviewData.values)}>
+          {filteredPreviewData.values.map(value =>
+            <Column key={value.value} header={this.getHeaderValue(value)}
+                    headerStyle={{textAlign: 'left', width: '5rem'}} style={{width: '5rem'}}
+                    field={value.value}/>
+          )}
+        </DataTable> :
+        <div style={styles.warningMessage}>
+          {filteredPreviewData.isLoading ?
+            <div>Generating preview for {domainDisplayed}</div> :
+            <div>No values found for {domainDisplayed}</div>
```

Comment:
This is likely an error case.

Reply:
No, because there is still a large chance of getting errors on this page. Just because the bigquery call can time out. That said, maybe the right thing to do instead of an isLoading is making an `isErrored` flag for that case? I don't have too many strong opinions there.

Reply:
The object-oriented scheme would be to have one abstract widget to go in this div with three subclasses for `loading`, `error`, and `loadedCorrectly`.

But yeah, if a domain has no values, I think that's a programming error, so we can throw a scary generic operation failed message and log an error. We should also have tests to verify that can't happen.